contracts/estate/EstateRegistry.sol

```diff
@@ -0,0 +1,236 @@
+      _pushTokenId(estateId, tokenId);
+    }
+    return bytes4(0xf0b9e5ba);
+  }
+
+  function ammendReceived(uint256 estateId, uint256 tokenId) external {
+    _pushTokenId(estateId, tokenId);
+  }
+
+  function transferToken(
+    uint256 estateId,
+    uint256 tokenId,
+    address destinatary
+  )
+    onlyAuthorized(estateId)
+    external
+  {
+    return _transferToken(estateId, tokenId, destinatary);
+  }
+
+  function transferManyTokens(uint256 estateId, uint256[] tokens, address destinatary) onlyAuthorized(estateId) external {
+    uint length = tokens.length;
+    for (uint i = 0; i < length; i++) {
+      _transferToken(estateId, tokens[i], destinatary);
+    }
+  }
+
+  function getTokenEstateId(uint256 tokenId) external view returns(uint256) {
+    return tokenIdEstate[tokenId];
+  }
+
+  function ping() onlyOwner external {
+    dar.ping();
+  }
+
+  function getSize(uint256 estateId) external view returns (uint256) {
+    return estateTokenIds[estateId].length;
+  }
+
+  function updateMetadata(uint256 estateId, string metadata) external onlyUpdateAuthorized(estateId) {
+    estateData[estateId] = metadata;
+  }
+
+  function getMetadata(uint256 estateId)
+    view
+    external
+    returns (string)
+  {
+    return estateData[estateId];
+  }
+
+  function setUpdateOperator(uint256 estateId, address operator) external onlyAuthorized(estateId) {
+    updateOperator[estateId] = operator;
+  }
+
+  // I don't like inverting the params here in terms of `setUpdateOperator`,
+  // but LANDRegistry does this, so might be best to be consistent
+  function isUpdateAuthorized(address operator, uint256 estateId) external view returns (bool) {
+    return _isUpdateAuthorized(operator, estateId);
+  }
+
+  function isAuthorized(address operator, uint256 estateId) external view returns (bool) {
+    return _isAuthorized(operator, estateId);
+  }
+
+  function _getNewTokenId() internal view returns (uint256) {
+    // This is not bullet-proof, but it'll be reinforced by the ERC721 uniqueness of the tokenId
+    // solium-disable-next-line security/no-block-members
+    return uint256(keccak256(abi.encodePacked(msg.sender, block.timestamp)));
+  }
+
+  function _pushTokenId(uint256 estateId, uint256 tokenId) internal {
+    require(estateTokenIndex[estateId][tokenId] == 0);
+    require(dar.ownerOf(tokenId) == address(this));
+
+    estateTokenIds[estateId].push(tokenId);
+
+    tokenIdEstate[tokenId] = estateId;
+
+    estateTokenIndex[estateId][tokenId] = estateTokenIds[estateId].length;
+  }
+
+  function _transferToken(
+    uint256 estateId,
+    uint256 tokenId,
+    address destinatary
+  )
+    internal
+  {
+    uint256[] storage tokenIds = estateTokenIds[estateId];
+    mapping(uint256 => uint256) tokenIndex = estateTokenIndex[estateId];
+
+    /**
+     * Using 1-based indexing to be able to make this check
+     */
+    require(tokenIndex[tokenId] != 0);
+
+    uint lastIndexInArray = tokenIds.length - 1;
```

Comment:
I think it is recommended for math operation to use [SafeMath](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/math/SafeMath.sol) even they seem to be simple & safe 

`sub(tokenIds.length, 1)`